patchwork-registries/src/main/java/net/minecraftforge/registries/ForgeRegistry.java

```diff
@@ -204,4 +272,72 @@ public int hashCode() {
 			return identifier.hashCode() * 33 + value.hashCode();
 		}
 	}
+
+	/**
+	 * Used to control the times where people can modify this registry.
+	 * Users should only ever register things in the Register<?> events!
+	 */
+	public void freeze() {
+		this.isFrozen = true;
+	}
+
+	public void unfreeze() {
+		this.isFrozen = false;
+	}
+
+	@Override
+	public boolean isLocked() {
+		return this.isFrozen;
+	}
+
+	@Override
+	public void clear() {
+		if (!this.isModifiable) {
+			throw new UnsupportedOperationException("Attempted to clear a non-modifiable Forge Registry");
+		}
+
+		if (this.isLocked()) {
+			throw new IllegalStateException("Attempted to clear the registry to late.");
+		}
+
+		if (!this.isVanilla) {
+			throw new UnsupportedOperationException("Vanilla registry does not support clear().");
+		}
+
+		if (this.clear != null) {
+			this.clear.onClear(this, stage);
+		}
+
+		ExtendedVanillaRegistry<V> reg = (ExtendedVanillaRegistry<V>) this.vanilla;
+		reg.clear();
+	}
+
+	@Override
+	public V remove(Identifier key) {
+		if (!this.isModifiable) {
+			throw new UnsupportedOperationException("Attempted to remove from a non-modifiable Forge Registry");
+		}
+
+		if (this.isLocked()) {
+			throw new IllegalStateException("Attempted to remove from the registry to late.");
+		}
+
+		if (!this.isVanilla) {
+			throw new UnsupportedOperationException("Vanilla registry does not support remove().");
+		}
+
+		throw new UnsupportedOperationException("Remove() is not implemented for Forge Mod registries.");
+	}
+
+	@SuppressWarnings("unchecked")
+	@Override
+	public <T> T getSlaveMap(Identifier name, Class<T> type) {
```

Comment:
ForgeRegistry#getSlaveMap and ForgeRegistry#setSlaveMap are for handing special vanilla case:
StructrueFeature extends Feature, this is not allowed in Forge, so they created "SlaveMap". Not sure if we should keep this.

Reply:
Is it something that could be redirected in Patcher, or is it too much of a hack?

Reply:
I'm also looking at it

Reply:
I think it's possible to work with this system while still using our vanilla wrapper implementation, I think we can avoid patcher here.

Reply:
Forge mods register both Feature and StructureFeature during IForgeRegistry<Feature<?>> registry event:
https://github.com/Cheaterpaul/Vampirism/blob/1.14/src/main/java/de/teamlapen/vampirism/core/ModFeatures.java

Reply:
getSlaveMap is declared in IForgeRegistry, so it is a part of the Forge API, we should keep this to maintain compatibility

Reply:
StructureFeature is supported